ethcore/builtin/src/lib.rs

```diff
@@ -193,15 +195,29 @@ impl ModexpPricer {
 ///
 /// Unless `is_active` is true,
 pub struct Builtin {
-	pricer: Box<dyn Pricer>,
-	native: Box<dyn Implementation>,
-	activate_at: u64,
+	pricer: BTreeMap<u64, Pricing>,
+	native: EthereumBuiltin,
 }
 
 impl Builtin {
 	/// Simple forwarder for cost.
+	///
+	/// Return the cost of the most recently activated pricer at the current block number.
+	///
+	/// If no pricer is actived `zero` is returned
+	///
+	/// If multiple `activation_at` has the same block number the last one is used
+	/// which follows the BTreeMap semantics
+	///
+	// NOTE(niklasad1): we could remove is_active and replace with `fn cost(...) -> Option<U256>`
+	// but it would require some changes that I don't want to do...
```

Comment:
I actually prefer the current approach since it's more explicit. So if we won't change that, let's remove the NOTE.